tsl/src/continuous_aggs/create.c

```diff
@@ -3021,17 +3096,29 @@ cagg_rebuild_view_definition(ContinuousAgg *agg, Hypertable *mat_ht)
 	fqi.finalized = finalized;
 	finalizequery_init(&fqi, direct_query, &mattblinfo);
 
-	mattablecolumninfo_addinternal(&mattblinfo);
+	if (!finalized)
+		mattablecolumninfo_addinternal(&mattblinfo);
```

Comment:
This is also part of existing code not something done in this PR.

Reply:
But the previous line did not have that check?

Reply:
You mean for mattablecomuninfo_init?
The same sequence of steps are done in cagg_create, the explicit comments are also there.
Basically we need internal column info only when not finalized, mattbl will there in any case.

Reply:
@RafiaSabih we've changed the behavior of the `cagg_rebuild_view_definition` because before it rebuild just then the view is in the old format (aka finalized=false). Now we enable rebuild the view definition for both finalized and not finalized caggs and also added a parameter to `force` the rebuild... so is good to add comments about it.